lib/chef/provider/apt_repository.rb

```diff
@@ -51,6 +51,19 @@ def load_current_resource
           end
         end
 
+        # make sure we have apt-transport-https installed before we try to
+        # apt-get update with a https repo defined
+        if new_resource.uri =~ /^https/
+          declare_resource(:apt_update, new_resource.name) do
+            ignore_failure true
+            action :periodic
+          end
```

Comment:
I see `apt-get update` fail on Travis all the time due to transient network oopses, I throw `retries 5` on basically everything that touches the network at this point because ¯\_(ツ)_/¯.